crates/cext/cbindgen.toml

```diff
@@ -33,6 +33,9 @@ typedef double complex QkComplex64;
 // Always expose [cfg(feature = "cbinding")] -- workaround for
 // https://github.com/mozilla/cbindgen/issues/995
 #define QISKIT_WITH_CBINDINGS
+
+typedef struct QkQuantumRegister QkQuantumRegister;
+typedef struct QkClassicalRegister QkClassicalRegister;
```

Comment:
Why do we need to define the Registers in this case through `cbindgen`? I was under the impression that `[export.rename]` would suffice.

Reply:
Since the register definitions are built out in a macro cbindgen is not able to detect the symbols. If you run cbindgen now (even with these lines) it emits a warning saying it can't find the definition of `ClassicalRegister` and `QuantumRegister`. There is an option to tell cbindgen to run `cargo expand` to build out the macros before doing the code scan to find the symbols but when I set that it led to a compilation error running cbindgen. Manually specifying the typedefs like this is the only way I could get a functional header with cbindgen.